automation/tests.py

```diff
@@ -311,8 +311,27 @@ def run_ktlint():
 def run_swiftlint():
     if on_darwin():
         run_command(['swiftlint', '--strict'])
+    elif not docker_installed():
+        print("WARNING: On non-Darwin hosts, docker is required to run swiftlint")
+        print("WARNING: skipping swiftlint on non-Darwin host")
     else:
-       print("WARNING: skipping swiftlint on non-Darwin host")
+        cwd = os.getcwd()
+
+        run_command(
+            [
+                "docker",
+                "run",
+                "-it",
+                "--rm",
+                "-v",
+                f"{cwd}:{cwd}",
+                "-w",
+                cwd,
+                "ghcr.io/realm/swiftlint:latest",
```

Comment:
Hmm do we know if this will always match the version of swiftlint running in CI (CI's version is not fixed, although I'm slightly in the camp of fixing the version - if we do that this will be fixed to that version which will be ok!)

Reply:
The version in CI is from `brew install swiftlint`, which should be the latest version. I also am pro fixing the verisons, esp because they seem to add new default lints in versions which causes PRs to fail for unrelated reasons :)